File: solutions/2088-minimum-time-to-type-word-using-special-typewriter/minimum-time-to-type-word-using-special-typewriter.swift
class Solution {
    func minTimeToType(_ word: String) -> Int {
        var ans = word.count
        var cur = 1
        for ch in word {
            let pos = Int(ch.asciiValue!) - 96
            let diff = abs(pos - cur)
            ans += min(diff, 26 - diff)
            cur = pos
        }
        return ans
    }
}

File: solutions/2088-minimum-time-to-type-word-using-special-typewriter/minimum-time-to-type-word-using-special-typewriter.rb
def min_time_to_type(word)
  ans = word.length
  cur = 1
  word.each_byte do |b|
    pos = b - 96
    diff = (pos - cur).abs
    ans += [diff, 26 - diff].min
    cur = pos
  end
  ans
end